sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -20,52 +24,133 @@ const double _kThumbRadius = 10.0;
 const double _kSwitchHeight = _kThumbRadius * 2.0;
 const double _kTrackHeight = 14.0;
 const double _kTrackRadius = _kTrackHeight / 2.0;
-const double _kTrackWidth = _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const double _kTrackWidth =
+    _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const Duration _kCheckDuration = const Duration(milliseconds: 200);
+const Size _kSwitchSize = const Size(_kSwitchWidth + 2.0, _kSwitchHeight + 2.0);
 
-class Switch extends Toggleable {
-  // TODO(jackson): Hit-test the switch so that it can respond to both taps and swipe gestures
+class Switch extends Component {
+  Switch({Key key, this.value, this.onChanged}) : super(key: key);
 
-  Switch({
-    Key key,
-    bool value,
-    ValueChanged onChanged
-  }) : super(key: key, value: value, onChanged: onChanged);
+  final bool value;
+  final ValueChanged onChanged;
 
-  Size get size => const Size(_kSwitchWidth + 2.0, _kSwitchHeight + 2.0);
+  Widget build() {
+    return new _SwitchWrapper(
+        value: value,
+        onChanged: onChanged,
+        thumbColor: Theme.of(this).accentColor);
+  }
+}
+
+// This wrapper class exists only because Switch needs to be a Component in
+// order to get an accent color from a Theme but Components do not know how to
+// host RenderObjects.
+class _SwitchWrapper extends OneChildRenderObjectWrapper {
```

Comment:
I wonder if you want to use LeafRenderObjectWrapper as the base class here given that you don't support having a child...  I'm not sure it matters much though.